src/meta/upgradeData/MetaDataUpgrade.cpp

```diff
@@ -0,0 +1,413 @@
+                                       const folly::StringPiece &val) {
+    auto oldItem = oldmeta::MetaServiceUtilsV1::parseIndex(val);
+    cpp2::IndexItem newItem;
+    newItem.set_index_id(oldItem.get_index_id());
+    newItem.set_index_name(oldItem.get_index_name());
+    cpp2::SchemaID schemaId;
+    if (oldItem.get_schema_id().getType() == oldmeta::cpp2::SchemaID::Type::tag_id) {
+        schemaId.set_tag_id(oldItem.get_schema_id().getType());
+    } else {
+        schemaId.set_edge_type(oldItem.get_schema_id().getType());
+    }
+    newItem.set_schema_id(schemaId);
+    NG_LOG_AND_RETURN_IF_ERROR(convertToNewColumns(oldItem.fields, newItem.fields));
+    NG_LOG_AND_RETURN_IF_ERROR(put(key, MetaServiceUtils::indexVal(newItem)));
+    return Status::OK();
+}
+
+Status MetaDataUpgrade::rewriteConfigs(const folly::StringPiece &key,
+                                       const folly::StringPiece &val) {
+    auto item = oldmeta::MetaServiceUtilsV1::parseConfigValue(val);
+
+    Value configVal;
+    switch (item.get_type()) {
+        case oldmeta::cpp2::ConfigType::INT64: {
+            auto value = *reinterpret_cast<const int64_t *>(item.get_value().data());
+            configVal.setInt(boost::get<int64_t>(value));
+            break;
+        }
+        case oldmeta::cpp2::ConfigType::DOUBLE: {
+            auto value = *reinterpret_cast<const double *>(item.get_value().data());
+            configVal.setFloat(boost::get<double>(value));
+            break;
+        }
+        case oldmeta::cpp2::ConfigType::BOOL: {
+            auto value = *reinterpret_cast<const bool *>(item.get_value().data());
+            configVal.setBool(boost::get<bool>(value) ? "True" : "False");
+            break;
+        }
+        case oldmeta::cpp2::ConfigType::STRING: {
+            configVal.setStr(boost::get<std::string>(item.get_value()));
+            break;
+        }
+        case oldmeta::cpp2::ConfigType::NESTED: {
+            auto value = item.get_value();
+            // transform to map value
+            conf::Configuration conf;
+            auto status = conf.parseFromString(boost::get<std::string>(value));
+            if (!status.ok()) {
+                LOG(ERROR) << "Parse value: " << value
+                           << " failed: " << status;
+                return Status::Error("Parse value: %s failed", value.c_str());
+            }
+            Map map;
+            conf.forEachItem(
+                    [&map](const folly::StringPiece&confKey, const folly::dynamic &confVal) {
+                map.kvs.emplace(confKey, confVal.asString());
+            });
+            configVal.setMap(std::move(map));
+            break;
+        }
+    }
+    auto newVal = MetaServiceUtils::configValue(
+            static_cast<cpp2::ConfigMode>(item.get_mode()), configVal);
+    NG_LOG_AND_RETURN_IF_ERROR(put(key, newVal));
+    return Status::OK();
+}
+
+Status MetaDataUpgrade::rewriteJobDesc(const folly::StringPiece &key,
```

Comment:
Maybe we could ignore them and just delete the job, 1.0 only has flush and compact, and they can retry another one in 2.0. WDYT? @liuyu85cn

Reply:
agree

Reply:
As discussed earlier, hope to provide the user with the jobs which they execute before.

Reply:
Almost meaningless, job manager will remove the expired jobs as well. And compact/flush job could retry.